src/System.Linq/src/System/Linq/Last.cs

```diff
@@ -107,49 +103,45 @@ internal static TSource TryGetLast<TSource>(this IEnumerable<TSource> source, Fu
             OrderedEnumerable<TSource> ordered = source as OrderedEnumerable<TSource>;
             if (ordered != null)
             {
-                return ordered.TryGetLast(predicate, out found);
+                bool found;
+                element = ordered.TryGetLast(predicate, out found);
+                return found;
             }
-
-            IList<TSource> list = source as IList<TSource>;
-            if (list != null)
+            else
             {
-                for (int i = list.Count - 1; i >= 0; --i)
+                IList<TSource> list = source as IList<TSource>;
+                if (list != null)
                 {
-                    TSource result = list[i];
-                    if (predicate(result))
+                    for (int i = list.Count - 1; i >= 0; --i)
                     {
-                        found = true;
-                        return result;
+                        TSource item = list[i];
+                        if (predicate(item))
+                        {
+                            element = item;
+                            return true;
+                        }
                     }
                 }
-            }
-            else
-            {
-                using (IEnumerator<TSource> e = source.GetEnumerator())
+                else
                 {
-                    while (e.MoveNext())
+                    TSource result = default(TSource);
+                    bool found = false;
```

Comment:
Nit: unneeded local variables. You can replace the `result = item; found = true` with `element = item; return true` and delete both of these variables.